apps/server/src/modules/user/repo/role.repo.integration.spec.ts

```diff
@@ -0,0 +1,295 @@
+import { NotFoundError, ValidationError } from '@mikro-orm/core';
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MongoMemoryDatabaseModule } from '../../database';
+import { Role } from '../entity';
+import { RoleRepo } from './role.repo';
+
+describe('role repo', () => {
+	let module: TestingModule;
+	let repo: RoleRepo;
+	let em: EntityManager;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [
+				MongoMemoryDatabaseModule.forRoot({
+					entities: [Role],
+				}),
+			],
+			providers: [RoleRepo],
+		}).compile();
+		repo = module.get(RoleRepo);
+		em = module.get<EntityManager>(EntityManager);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('entity', () => {
+		it.skip('should fail for double creating a uniuqe role name.', async () => {
+			const nameA = `a${Date.now()}`;
+			const roleA1 = em.create(Role, { name: nameA });
+			await em.persistAndFlush([roleA1]);
+			const roleA2 = em.create(Role, { name: nameA });
+
+			await expect(em.persistAndFlush([roleA2])).rejects.toThrow(ValidationError);
+		});
+
+		it.skip('should fail for double creating a uniuqe role name in same step', async () => {
+			const nameA = `a${Date.now()}`;
+			const roleA1 = em.create(Role, { name: nameA });
+			const roleA2 = em.create(Role, { name: nameA });
+
+			await expect(em.persistAndFlush([roleA1, roleA2])).rejects.toThrow(ValidationError);
+		});
+
+		it.todo('should fail if permission by creating is added, that not exist as enum.');
+	});
+
+	describe('findByName', () => {
+		afterEach(async () => {
+			repo.cl1.clear();
+			await em.nativeDelete(Role, {});
+		});
+
+		it('should return right keys', async () => {
+			const nameA = `a${Date.now()}`;
+			const roleA = em.create(Role, { name: nameA });
+
+			await em.persistAndFlush([roleA]);
+			const result = await repo.findByName(nameA);
+			expect(Object.keys(result).sort()).toEqual(
+				['createdAt', 'updatedAt', 'permissions', 'roles', 'name', '_id'].sort()
+			);
+		});
+
+		it('should return one role that matched by name', async () => {
+			const nameA = `a${Date.now()}`;
+			const nameB = `b${Date.now()}`;
+			const roleA = em.create(Role, { name: nameA });
+			const roleB = em.create(Role, { name: nameB });
+
+			await em.persistAndFlush([roleA, roleB]);
+			const result = await repo.findByName(nameA);
+			expect(result).toEqual(roleA);
+		});
+
+		it('should throw an error if roles by name not exist', async () => {
+			const nameA = `a${Date.now()}`;
+			const nameB = `b${Date.now()}`;
+			const roleA = em.create(Role, { name: nameA });
+
+			await em.persistAndFlush([roleA]);
```

Comment:
why create a role here? might be enough to simply define a name that is unique (timestamp) and search for it...

Reply:
true